src/hal/lpc17xx/layout.ld

```diff
@@ -10,7 +10,7 @@ ENTRY(main)
 
 MEMORY
 {
-    rom(RX)   : ORIGIN = 0x00000000, LENGTH = 0x8000
+    rom(RX)   : ORIGIN = 0x00000000, LENGTH = 64K
```

Comment:
Here's the [parametric search for the LPC17xx](http://www.nxp.com/parametrics/1403790745385/#/p=1,s=0,f=,c=,rpp=,fs=0,sc=c4c8d2,so=asc,es=), sorted by Flash size.  It looks like the smallest is the LPC1751FBD80 with 32KB of flash.  I can probably change it to be that more conservative value and still have plenty of flash space for any of the examples, even with debug.